src/main/java/Parser.java

```diff
@@ -0,0 +1,330 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Reads and processes the commands given to Duke and returns the appropriate action.
+ * Generates either todo, deadline or event tasks.
+ */
+public class Parser {
+    private TaskList taskList;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Creates a Parser object with the given TaskList, Ui, Storage.
+     * @param taskList TaskList that duke keeps track of.
+     * @param ui Generates the output string.
+     * @param storage Accesses the given file.
+     */
+    public Parser(TaskList taskList, Ui ui, Storage storage) {
+        this.taskList = taskList;
+        this.ui = ui;
+        this.storage = storage;
+    }
+
+    /**
+     * Processes the Command given by the user to Duke.
+     * @param command String representing the command given to Duke.
+     * @return A string representing the reply from Duke.
+     */
+    public String processCommand(String command) {
+        String reply = "";
+
+        // When command is 'list'
+        if (command.equals("list")) {
+            String list = taskList.printAllTasks();
+            reply = "Here are the tasks in your list:" + "\n" + list;
+        } else if (command.contains("done")) {
+            reply = executeDone(command);
+        } else if (command.contains("delete")) {
+            reply = executeDelete(command);
+        } else if (command.contains("find")) {
+            reply = executeFind(command);
+        } else if (containsTask(command)) {
+            // Processes task command, not the same as generateNewTask()
+            reply = executeNewTask(command);
+        } else {
+            // Unknown command
+            reply = ui.showErrorMessage(new UnknownTaskTypeException());
+        }
+
+        // When command is bye
+        if (command.equals("bye")) {
+            reply = ui.showFarewell();
+        }
+
+        return reply;
+    }
+
+
+
+    /**
+     * Generates the appropriate task that matches the command.
+     * @param task command to create the task.
+     * @return Task that is generated.
+     * @throws DukeException Exceptions specific to Duke
+     */
+    public static Task generateNewTask(String task) throws DukeException {
+        try {
+            String type = task.substring(0, task.indexOf(' '));
+            String taskDescription = task.substring(task.indexOf(' ') + 1);
+            boolean hasTags = false;
+            String tag = "";
+
+            if (taskDescription.contains("#")) {
+                hasTags = true;
+                String[] processed = taskDescription.split("#");
+                taskDescription = processed[0];
+                tag = processed[1];
+            }
+
+            Task newTask = new Task("dummy");
+
+            // Create the appropriate Task type
+            if (type.equals("todo")) {
+                newTask = new ToDo(taskDescription);
+            }
+
+            try {

```

Comment:
Is there a reason why you you used multiple consecutive try-except blocks that are executed right after each other to create tasks? The code inside each of them are very similar and should be abstracted out into a function if possible. Moreover, it's possible to catch and handle multiple exception in one try-catch block, which makes reading the code easier